bpf/lib/nodeport.h

```diff
@@ -986,15 +1022,26 @@ static __always_inline int nodeport_nat_ipv4_fwd(struct __ctx_buff *ctx)
 
 #ifdef ENABLE_DSR
 #if DSR_ENCAP_MODE == DSR_ENCAP_IPIP
+static __always_inline __be32 rss_gen_src4(__be32 client, __be32 l4_hint)
+{
+	const __u32 bits = 32 - IPV4_RSS_PREFIX_BITS;
+	__be32 src = IPV4_RSS_PREFIX;
+
+	if (bits)
+		src |= bpf_htonl(hash_32(client ^ l4_hint, bits));
```

Comment:
Why are we passing a `__be32` into `htonl()`?

Reply:
You mean the `client ^ l4_hint`? It doesn't really matter given this is just input for the hash to calc in this case.